### ISSUE TITLE
Bugs found by static tests

### ISSUE THREAD

**Issue by tester7632** [openssl/openssl#11918]
Running the latest git version 2de6466 through sonarqube revealed some potential bugs:

https://sonarcloud.io/project/issues?id=tester7632_openssl&resolved=false&severities=BLOCKER&types=BUG

There are quite a number of "bugs" reported by the tool, but as far as I can see most are only "not best practice" kind of bugs and not worth the time to fix them. The one listed as BLOCKER look like they should get a closer look though.

Complete report:
https://sonarcloud.io/dashboard?id=tester7632_openssl

**Comment by kroeckx** [comment]
This looks like yet an other tool with 99% false positives.

**Comment by richsalz** [comment]
I looked at one of the "blocker" items, https://sonarcloud.io/project/issues?id=tester7632_openssl&open=AXI9Kb9LAi29hCJ8RXj4&resolved=false&severities=BLOCKER&types=BUG.  It is wrong, complaining about a condition that can't happen.

The project uses Coverity static scanning.  I think someone will have to review all of the issues here and open an issue for the ones that are valid.  I doubt the team is interested in doing this.

**Comment by kroeckx** [comment]
You really randomly looked at the same issue I did? :)

**Comment by richsalz** [comment]
I looked at code that I originally wrote :)  And I didn't see your comment until after I posted mine.

**Comment by tester7632** [comment]
Yes this tool reports a lot of false positives and everything that is not "best practice" 
Thats why I only pointed out the blocker items as I can not reliably tell if everything listed there is bogus and in the case it is not ...

**Comment by BenBE** [comment]
Just had a closer look at
https://sonarcloud.io/project/issues?id=tester7632_openssl&open=AXI9KcWbAi29hCJ8RZQN&resolved=false&severities=BLOCKER&types=BUG
(OOB memory access in `opt_help`)
which is a OOB stack write after the array `start` if `MAX_OPT_HELP_WIDTH > 80` (plus some other conditions). Not an immediate issue AFAICS, as the passed in data at the point seems to be constant.

HAven't looked at any other of the reports, but from my experience, once those tools can tell you some consistent steps to follow along, there's usually also some minor detail, that the code misses. In this case, the static buffer size for `start`.

**Comment by nhorman** [comment]
Marking as inactive, to be closed at the end of 3.4 dev, barring further input

**Comment by t8m** [comment]
Closing as the issue was labeled as inactive for extended period of time and there was no additional input.
Please add a comment if you have any further information to support reopening the issue.
